### PR TITLE
feat: add python3.11 and dependencies

### PR DESCRIPTION
**Pull request by hartmutobendorf**
Added python3.11 and its dependencies for 22.04.

Removed file patterns that were no longer used for libpython.

### CODE REVIEW

File: slices/libpython3.11-minimal.yaml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      /usr/lib/python3.11/subprocess.py:
+      /usr/lib/python3.11/sysconfig.py:
+      /usr/lib/python3.11/tempfile.py:
+      /usr/lib/python3.11/textwrap.py:
+      /usr/lib/python3.11/threading.py:
+      /usr/lib/python3.11/token.py:
+      /usr/lib/python3.11/tokenize.py:
+      /usr/lib/python3.11/traceback.py:
+      /usr/lib/python3.11/tracemalloc.py:
+      /usr/lib/python3.11/types.py:
```
Comment (rebornplusplus) [inline slices/libpython3.11-minimal.yaml:105]: `/usr/lib/python3.11/typing.py` file seems to be missing.

File: slices/libpython3.11-minimal.yaml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      /usr/lib/python3.11/configparser.py:
+      /usr/lib/python3.11/contextlib.py:
+      /usr/lib/python3.11/copy.py:
+      /usr/lib/python3.11/copyreg.py:
+      /usr/lib/python3.11/csv.py:
+      /usr/lib/python3.11/datetime.py:
+      /usr/lib/python3.11/dis.py:
+      /usr/lib/python3.11/email/**:
+      /usr/lib/python3.11/encodings/**:
+      /usr/lib/python3.11/enum.py:
```
Comment (rebornplusplus) [inline slices/libpython3.11-minimal.yaml:49]: `/usr/lib/python3.11/filecmp.py` seems to be missing.

File: slices/libpython3.11-stdlib.yaml
Comment (rebornplusplus) [inline slices/libpython3.11-stdlib.yaml:1]: I think some new libraries in Python3.11 are not included in this file.

To help you out, here's a diff of the paths between [libpython3.8-stdlib](https://packages.ubuntu.com/focal/amd64/libpython3.8-stdlib/filelist) and [libpython3.11-stdlib](https://packages.ubuntu.com/jammy-updates/amd64/libpython3.11-stdlib/filelist):

```diff
1c1,5
< /usr/lib/python3.8/__phello__.foo.py
---
> /usr/lib/python3.11/__hello__.py
> /usr/lib/python3.11/__phello__/__init__.py
> /usr/lib/python3.11/__phello__/spam.py
> /usr/lib/python3.11/_aix_support.py
> /usr/lib/python3.11/_bootsubprocess.py
3c7
< /usr/lib/python3.8/_dummy_thread.py
---
> /usr/lib/python3.11/_distutils_system_mod.py
25a30
> /usr/lib/python3.11/asyncio/mixins.py
34a40
> /usr/lib/python3.11/asyncio/taskgroups.py
35a42,43
> /usr/lib/python3.11/asyncio/threads.py
> /usr/lib/python3.11/asyncio/timeouts.py
43d50
< /usr/lib/python3.8/binhex.py
80,81d86
< /usr/lib/python3.8/dummy_threading.py
< /usr/lib/python3.8/filecmp.py
83d87
< /usr/lib/python3.8/formatter.py
87a92
> /usr/lib/python3.11/graphlib.py
133a139
> /usr/lib/python3.11/lib-dynload/_typing.cpython-311-x86_64-linux-gnu.so
136a143
> /usr/lib/python3.11/lib-dynload/_zoneinfo.cpython-311-x86_64-linux-gnu.so
141d147
< /usr/lib/python3.8/lib-dynload/parser.cpython-311-x86_64-linux-gnu.so
145a152
> /usr/lib/python3.11/lib-dynload/xxlimited_35.cpython-311-x86_64-linux-gnu.so
209d215
< /usr/lib/python3.8/symbol.py
231a238,243
> /usr/lib/python3.11/test/support/bytecode_helper.py
> /usr/lib/python3.11/test/support/hashlib_helper.py
> /usr/lib/python3.11/test/support/import_helper.py
> /usr/lib/python3.11/test/support/interpreters.py
> /usr/lib/python3.11/test/support/logging_helper.py
> /usr/lib/python3.11/test/support/os_helper.py
232a245
> /usr/lib/python3.11/test/support/socket_helper.py
233a247,248
> /usr/lib/python3.11/test/support/threading_helper.py
> /usr/lib/python3.11/test/support/warnings_helper.py
236a252,255
> /usr/lib/python3.11/tomllib/__init__.py
> /usr/lib/python3.11/tomllib/_parser.py
> /usr/lib/python3.11/tomllib/_re.py
> /usr/lib/python3.11/tomllib/_types.py
240d258
< /usr/lib/python3.8/typing.py
242a261
> /usr/lib/python3.11/unittest/_log.py
265a285
> /usr/lib/python3.11/wsgiref/types.py
294a315,318
> /usr/lib/python3.11/zoneinfo/__init__.py
> /usr/lib/python3.11/zoneinfo/_common.py
> /usr/lib/python3.11/zoneinfo/_tzpath.py
> /usr/lib/python3.11/zoneinfo/_zoneinfo.py
```

But there does not seem to be adequate diff in the slice definition file (from 3.8) to accommodate the new libraries.

```diff
18c18
<       - mime-support_mime-types
---
>       - media-types_data
42d41
<       /usr/lib/python3.8/typing.py:
51c50
<       - libffi7_libs
---
>       - libffi8_libs
69d67
<       /usr/lib/python3.8/_dummy_thread.py:
71d68
<       /usr/lib/python3.8/dummy_threading.py:
141c138
<       /usr/lib/python3.8/__phello__.foo.py:
---
>       /usr/lib/python3.11/__phello__/**:
183c180
<       /usr/lib/python3.8/filecmp.py:
---
> #      /usr/lib/python3.11/filecmp.py:
256d252
<       /usr/lib/python3.8/lib-dynload/parser.cpython-311-*-linux-*.so:
259d254
<       /usr/lib/python3.8/symbol.py:
295d289
<       /usr/lib/python3.8/binhex.py:
306d299
<       - libmpdec2_libs
382,383c375
<       /usr/lib/python3.8/formatter.py:
<       /usr/lib/python3.8/this.py:
---
>       /usr/lib/python3.11/this.py:
\ No newline at end of file
```

**Review by rebornplusplus — CHANGES_REQUESTED** [review]
Hi! Nice chiselling there. It's looking good.

I left a few comments below after a quick scan through the files. I will be thorough in the next pass. Here are a few general comments which I think might improve this PR:

* I think it is better to sort the entries in `essential` and `contents` in the slice definition files. I agree it hasn't been done so for the existing files in the repo. But I think it would be nice to do that from now on.
* nit: There are no newlines at the end of most files.

File: slices/python3.11-minimal.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package: python3.11-minimal
+
+slices:
+  bins:
+    essential:
+      - libc6_libs
```
Comment (cjdcordeiro) [inline slices/python3.11-minimal.yaml:6]: why is libc6 a dependency here? It is already a dependency of libexpat1 and zlib1g, so this is redundant

File: slices/readline-common.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package: readline-common
+
+slices:
+  config:
+    contents:
+      /etc/inputrc: { copy: /usr/share/readline/inputrc }
```
Comment (cjdcordeiro) [inline slices/readline-common.yaml:6]:
```suggestion
      /etc/inputrc: { copy: /usr/share/readline/inputrc }

```

File: slices/libreadline8.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+      - libtinfo6_libs
+      - readline-common_config
+    contents:
+      /lib/*-linux-*/libreadline.so.8*:
+      /lib/*-linux-*/libhistory.so.8*:
```
Comment (cjdcordeiro) [inline slices/libreadline8.yaml:11]:
```suggestion
      /lib/*-linux-*/libhistory.so.8*:

```

File: slices/libpython3.11-stdlib.yaml [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+
+  # Outliers and Deprecated Modules
+  # The "extra" slice consists of easter-eggs and deprecated modules
+  extras:
+    essential:
+      - libpython3.11-stdlib_core
+      - libpython3.11-stdlib_internet
+    contents:
+      /usr/lib/python3.11/antigravity.py:
+      /usr/lib/python3.11/this.py:
```
Comment (cjdcordeiro) [inline slices/libpython3.11-stdlib.yaml:375]:
```suggestion
      /usr/lib/python3.11/this.py:

```

File: slices/libpython3.11-minimal.yaml [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+      /usr/lib/python3.11/tokenize.py:
+      /usr/lib/python3.11/traceback.py:
+      /usr/lib/python3.11/tracemalloc.py:
+      /usr/lib/python3.11/types.py:
+      /usr/lib/python3.11/urllib/**:
+      /usr/lib/python3.11/uu.py:
+      /usr/lib/python3.11/warnings.py:
+      /usr/lib/python3.11/weakref.py:
+      /usr/lib/python3.11/zipfile.py:
+      /usr/lib/python3.11/zipimport.py:
```
Comment (cjdcordeiro) [inline slices/libpython3.11-minimal.yaml:111]:
```suggestion
      /usr/lib/python3.11/zipimport.py:

```

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
There are indeed some files missing. I'll confirm and defer to @rebornplusplus comments on that.

@hartmutobendorf we'd love to have your contribution merged :) do you think you'll be able to address the change requests? If not, we might take over the PR sometime during the next couple of weeks. Please let us know ;)

**Comment by rebornplusplus** [comment]
Hello @hartmutobendorf. Since we have not heard from you in a while, I have opened a new PR for Python3.11 slices: https://github.com/canonical/chisel-releases/pull/82. Thanks for starting this off, your work helped a lot for the new PR. I will be closing this PR, but please feel free to review the other one. I have added you as an optional reviewer.

**Pull request closed without merging**